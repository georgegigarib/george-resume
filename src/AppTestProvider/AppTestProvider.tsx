import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
