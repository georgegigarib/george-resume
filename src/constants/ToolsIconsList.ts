import JsIcon from '@/assets/icons/tools/Javascript.svg'

export type ToolInfo = {
  name: string
  icon: IconName
  color: string
  bgColor: string
  darkBgColor?: string
  borderColor: string
  darkBorderColor?: string
  darkColor?: string
  svg?: string
  custom?: boolean
}

export type ToolIconProps = {
  tool: IconName
  size?: '5x' | '4x' | '3x' | '2x' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

export enum IconName {
  React = 'react',
  ReactNative = 'reactnative',
  TailwindCSS = 'tailwindcss',
  JavaScript = 'js',
  TypeScript = 'ts-circle',
  VueJs = 'vuejs',
  NextJS = 'nextjs',
  NodeJS = 'nodejs',
  HTML = 'html',
  CSS = 'css',
  Express = 'express',
  Spotify = 'spotify',
  Git = 'git',
  Redis = 'redis',
  Bun = 'bun',
  Docker = 'docker',
  Pinia = 'pinia',
  Sass = 'sass',
  Bootstrap = 'bootstrap',
  Angular = 'angular',
  C = 'c',
  CSharp = 'csharp',
  Django = 'django',
  FramerMotion = 'framer-motion',
  Go = 'go',
  Graphql = 'graphql',
  Javalanguage = 'java',
  Laravel = 'laravel',
  Mui = 'mui',
  Jest = 'jest',
  Vitest = 'vitest',
  Cypress = 'cypress',
  MySQL = 'mysql',
  MongoDb = 'mongodb',
  Php = 'php',
  Postgresql = 'postgresql',
  Python = 'python',
  Redux = 'redux',
  Zustand = 'zustand',
  Svelte = 'svelte',
}

export const toolsIconsList: ToolInfo[] = [
  {
    name: 'React',
    icon: IconName.React,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'React',
    icon: IconName.ReactNative,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Tailwind CSS',
    icon: IconName.TailwindCSS,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#38BDF8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'JavaScript',
    icon: IconName.JavaScript,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
    custom: true,
    svg: JsIcon,
  },
  {
    name: 'Jest',
    icon: IconName.Jest,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MySQL',
    icon: IconName.MySQL,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#38BDF8',
  },
  {
    name: 'Vitest',
    icon: IconName.Vitest,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
  },
  {
    name: 'Cypress',
    icon: IconName.Cypress,
    color: '#000000FF',
    bgColor: '#B4B4B4FF',
    darkBgColor: '#BABABAFF',
    borderColor: '#C7B008FF',
  },
  {
    name: 'TypeScript',
    icon: IconName.TypeScript,
    color: '#3178C6',
    bgColor: '#FAFAFAFF',
    borderColor: '#3178C6',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Next.js',
    icon: IconName.NextJS,
    color: '#0000000',
    bgColor: '#FAFAFAFF',
    borderColor: '#555555FF',
    darkBgColor: '#3E3E3EFF',
    darkColor: '#FFFFFF',
    darkBorderColor: '#BFBFBFFF',
  },
  {
    name: 'Node.js',
    icon: IconName.NodeJS,
    color: '#539E43',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#539E43',
  },
  {
    name: 'HTML',
    icon: IconName.HTML,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'CSS',
    icon: IconName.CSS,
    color: '#264DE4',
    bgColor: '#FAFAFAFF',
    borderColor: '#264DE4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Express',
    icon: IconName.Express,
    color: '#333333',
    bgColor: '#FAFAFAFF',
    borderColor: '#CCCCCC',
  },
  {
    name: 'Spotify',
    icon: IconName.Spotify,
    color: '#1DB954',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#1DB954',
  },
  { name: 'Git', icon: IconName.Git, color: '#F05032', bgColor: '#F5F5F5', borderColor: '#F05032' },
  {
    name: 'Redis',
    icon: IconName.Redis,
    color: '#D82C20',
    bgColor: '#FAFAFAFF',
    borderColor: '#D82C20',
  },
  {
    name: 'Bun',
    icon: IconName.Bun,
    darkColor: '#ffffff',
    color: '#333333',
    bgColor: '#FAFAFAFF',
    darkBorderColor: '#bababa',
    borderColor: '#686767FF',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Docker',
    icon: IconName.Docker,
    color: '#2496ED',
    bgColor: '#FAFAFAFF',
    borderColor: '#2496ED',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Pinia',
    icon: IconName.Pinia,
    color: '#F0C723FF',
    darkBgColor: '#2C2C2C',
    bgColor: '#FAFAFAFF',
    borderColor: '#FFD939',
  },
  {
    name: 'Sass',
    icon: IconName.Sass,
    color: '#CC6699',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#CC6699',
  },
  {
    name: 'Bootstrap',
    icon: IconName.Bootstrap,
    color: '#A877F3FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#7952B3',
  },
  {
    name: 'Angular',
    icon: IconName.Angular,
    color: '#DD0031',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#DD0031',
  },
  {
    name: 'C',
    icon: IconName.C,
    color: '#00599C',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#00599C',
  },
  {
    name: 'C#',
    icon: IconName.CSharp,
    color: '#239120',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#239120',
  },
  {
    name: 'Django',
    icon: IconName.Django,
    color: '#092E20',
    bgColor: '#FAFAFAFF',
    borderColor: '#092E20',
  },
  {
    name: 'Framer Motion',
    icon: IconName.FramerMotion,
    color: '#EC5990',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#EC5990',
  },
  {
    name: 'Go',
    icon: IconName.Go,
    color: '#00ADD8',
    bgColor: '#FAFAFAFF',
    borderColor: '#00ADD8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'GraphQL',
    icon: IconName.Graphql,
    color: '#E10098',
    bgColor: '#FAFAFAFF',
    borderColor: '#E10098',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'java',
    icon: IconName.Javalanguage,
    color: '#007396',
    bgColor: '#FAFAFAFF',
    borderColor: '#007396',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Laravel',
    icon: IconName.Laravel,
    color: '#FF2D20',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF2D20',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Material UI',
    icon: IconName.Mui,
    color: '#0081CB',
    bgColor: '#FAFAFAFF',
    borderColor: '#0081CB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MongoDB',
    icon: IconName.MongoDb,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Vue Js',
    icon: IconName.VueJs,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PHP',
    icon: IconName.Php,
    color: '#959AE1FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#777BB4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PostgreSQL',
    icon: IconName.Postgresql,
    color: '#336791',
    bgColor: '#FAFAFAFF',
    borderColor: '#336791',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Python',
    icon: IconName.Python,
    color: '#3776AB',
    bgColor: '#FAFAFAFF',
    borderColor: '#3776AB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Redux',
    icon: IconName.Redux,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Zustand',
    icon: IconName.Zustand,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Svelte',
    icon: IconName.Svelte,
    color: '#FF3E00',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF3E00',
    darkBgColor: '#2C2C2C',
  },
]
