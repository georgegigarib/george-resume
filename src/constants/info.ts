export const WORK_EMAIL = 'georgegil.work@gmail.com'
export const SOCIAL_LINKS = {
  github: 'https://github.com/georgegigarib',
  linkedin: 'https://www.linkedin.com/in/georgegg/',
  instagram: 'URL_ADDRESS.com/GeorgeGil',
}
