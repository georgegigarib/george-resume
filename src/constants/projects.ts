import image1 from '@/assets/images/projects/movie-app.gif'
import image2 from '@/assets/images/projects/mobile-2.jpeg'
import image3 from '@/assets/images/projects/mobile-1.jpg'
import image4 from '@/assets/images/projects/desktop-app-1.png'
import image5 from '@/assets/images/projects/desktop-app-2.png'
import image6 from '@/assets/images/projects/desktop-app-3.png'
import { IconName } from './ToolsIconsList'

export type ProjectInfoType = {
  src: string
  title: string
  description: string
  url: string
  tools: IconName[]
}

export const mobileProjects: ProjectInfoType[] = [
  {
    src: image1,
    title: 'Movie App',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto y debo alargarlo mas y mas y mas',
    url: 'https://movie-app.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.JavaScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS, IconName.JavaScript]
  },
  {
    src: image3,
    title: 'Proyecto Mobile 2',
    description:
      ' Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript, IconName.JavaScript]
  },
  {
    src: image1,
    title: 'Proyecto Mobile 2',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description: 'Proyecto con animaciones personalizadas.',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS]
  }
]

export const desktopProjects: ProjectInfoType[] = [
  {
    src: image6,
    title: 'Proyecto Desktop 1',
    description: 'Dashboard avanzado.',
    url: 'https://desktop1.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description:
      'Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image4,
    title: 'Proyecto Desktop 3',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.NextJS, IconName.NodeJS, IconName.TailwindCSS, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  }
]
