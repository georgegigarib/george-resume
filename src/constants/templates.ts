import { IconName } from './ToolsIconsList'

export interface Template {
  tools: IconName[]
  url: string
  hint: string
  bgColor: string
  description: string
}

export const devTemplates: Template[] = [
  {
    tools: [IconName.TypeScript, IconName.React, IconName.TailwindCSS, IconName.Bun, IconName.CSS],
    url: 'https://github.com/example/fast-render-template',
    hint: 'fast rendering',
    bgColor: '#919469FF',
    description: 'A lightweight template designed for fast rendering with modern tools.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.NextJS,
      IconName.TailwindCSS,
      IconName.Docker,
      IconName.React
    ],
    url: 'https://github.com/example/next-ssr-template',
    hint: 'server rendering',
    bgColor: '#B99E4CFF',
    description: 'A server-side rendering starter template using Next.js and Docker.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.VueJs,
      IconName.TailwindCSS,
      IconName.Pinia,
      IconName.Docker
    ],
    url: 'https://github.com/example/vue-pinia-template',
    hint: 'vue simplicity',
    bgColor: '#6E8C9AFF',
    description: 'A clean Vue.js starter with Pinia and TailwindCSS for simplicity and scalability.'
  },
  {
    tools: [IconName.JavaScript, IconName.Angular, IconName.Sass, IconName.Docker, IconName.NodeJS],
    url: 'https://github.com/example/angular-enterprise-template',
    hint: 'enterprise apps',
    bgColor: '#A26B5AFF',
    description: 'A robust template for building enterprise-grade Angular applications.'
  },
  {
    tools: [IconName.TypeScript, IconName.React, IconName.Mui, IconName.Graphql, IconName.Django],
    url: 'https://github.com/example/react-graphql-template',
    hint: 'full stack',
    bgColor: '#877F95FF',
    description: 'Full-stack development made easy with React, MUI, and GraphQL backend.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.Svelte,
      IconName.TailwindCSS,
      IconName.Python,
      IconName.Postgresql
    ],
    url: 'https://github.com/example/svelte-minimal-template',
    hint: 'minimalist setup',
    bgColor: '#687969FF',
    description:
      'A minimalist yet powerful template using Svelte and Python with a PostgreSQL database.'
  },
  {
    tools: [IconName.JavaScript, IconName.React, IconName.Redux, IconName.Bootstrap, IconName.Php],
    url: 'https://github.com/example/ecommerce-template',
    hint: 'ecommerce base',
    bgColor: '#908C63FF',
    description: 'A ready-to-use base for building eCommerce solutions with React and PHP.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.Graphql,
      IconName.MongoDb,
      IconName.Docker
    ],
    url: 'https://github.com/example/data-intensive-template',
    hint: 'data intensive',
    bgColor: '#96659FFF',
    description: 'A powerful template for data-intensive applications using MongoDB and GraphQL.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.TailwindCSS,
      IconName.FramerMotion,
      IconName.NodeJS
    ],
    url: 'https://github.com/example/firebase-react-template',
    hint: 'real-time apps',
    bgColor: '#3E848DFF',
    description: 'A template for real-time apps with Firebase backend and React frontend.'
  }
]
