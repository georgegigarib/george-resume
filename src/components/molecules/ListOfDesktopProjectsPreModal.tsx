import React, { useRef, useState } from 'react'
import { Swiper, SwiperSlide } from 'swiper/react'
import { Pagination, Autoplay } from 'swiper/modules'
import { desktopProjects } from '@/constants/projects'
import '@/styles/projects-modal.css'

import 'swiper/css'
import 'swiper/css/pagination'
import type { Swiper as SwiperType } from 'swiper/types'
import { useTranslation } from 'react-i18next'
import { Tooltip } from '@mui/material'

const ListOfDesktopProjectsPreModal: React.FC = () => {
  const [activeIndex, setActiveIndex] = useState(0)
  const { t } = useTranslation()
  const toolsLimit = 7
  const progressCircle = useRef<SVGSVGElement>(null)
  const progressContent = useRef<HTMLSpanElement>({} as HTMLSpanElement)
  const onAutoplayTimeLeft = (_: SwiperType, time: number, progress: number) => {
    progressCircle.current!.style.setProperty('--progress', (1 - progress).toString())
    progressContent.current.textContent = `${Math.ceil(time / 1000)}s`
  }

  return (
    <div
      className="grid ml-8 grid-rows-1 grid-cols-4 
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
     border-l-0 rounded-l-none
    rounded-xl"
    >
      <div
        className="text-center mb-4 col-span-1 self-center p-2 mr-4 
        md:dark:bg-app-lightModeBoxes-box1 
    md:bg-white
    md:bg-opacity-20
    md:border-[2px]
    border-app-darkModeBoxes-box1
    dark:border-white
    dark:border-opacity-60
    md:shadow-md
    opacity-90 rounded-xl flex flex-col gap-4 mt-16"
      >
        <h3 className="hidden md:flex text-lg font-bold dark:text-white text-black">
          {t(desktopProjects[activeIndex]?.title)}
        </h3>

        <div className="hidden md:flex flex-wrap justify-center gap-2">
          {desktopProjects[activeIndex].tools.map((tool, index) =>
            index < toolsLimit ? (
              <Tooltip title={tool} arrow key={index}>
                <i className={`ci ci-${tool} ci-lg`}></i>
              </Tooltip>
            ) : index === toolsLimit ? (
              <Tooltip
                key={index}
                title={t('dev.projects.toolsRemaining', {
                  count: desktopProjects[activeIndex].tools.length - toolsLimit,
                })}
                arrow
              >
                <p className="-mt-1 dark:text-white text-black">
                  +{desktopProjects[activeIndex].tools.length - toolsLimit}
                </p>
              </Tooltip>
            ) : (
              <span key={index} />
            )
          )}
        </div>
      </div>

      <div className="col-span-3 self-center py-3 pr-3 ml-3 group">
        <Swiper
          initialSlide={2}
          grabCursor={true}
          spaceBetween={8}
          slidesPerView={1}
          slidesPerGroup={1}
          onAutoplayTimeLeft={onAutoplayTimeLeft}
          loop={true}
          autoplay={{
            pauseOnMouseEnter: true,
            delay: 3000,
          }}
          modules={[Autoplay, Pagination]}
          pagination={true}
          direction="vertical"
          className="h-[120px] md:h-[250px] w-full"
          onSlideChange={(swiper) => setActiveIndex(swiper.realIndex)}
        >
          {desktopProjects.map((project, index) => (
            <SwiperSlide key={index} className="bg-transparent">
              <img src={project.src} alt={project.title} className="rounded-2xl" />
            </SwiperSlide>
          ))}
          <div className="hidden md:block">
            <div className="autoplay-progress" slot="container-end">
              <svg viewBox="0 0 48 48" ref={progressCircle}>
                <circle cx="24" cy="24" r="20"></circle>
              </svg>
              <span ref={progressContent}></span>
            </div>
          </div>
        </Swiper>
      </div>
    </div>
  )
}

export default ListOfDesktopProjectsPreModal
