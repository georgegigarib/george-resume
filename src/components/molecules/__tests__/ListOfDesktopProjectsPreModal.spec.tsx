import { render, screen } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import ListOfDesktopProjectsPreModal from '@/components/molecules/ListOfDesktopProjectsPreModal'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { desktopProjects } from '@/constants/projects'

vi.mock('@/constants/projects', () => ({
  desktopProjects: [
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop1',
      tools: ['React', 'TypeScript'],
      src: 'desktop1.png',
    },
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop2',
      tools: ['React', 'TypeScript'],
      src: 'desktop2.png',
    },
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop3',
      tools: ['React', 'TypeScript'],
      src: 'desktop3.png',
    },
  ],
}))

describe.skip('ListOfDesktopProjectsPreModal Component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <ListOfDesktopProjectsPreModal />
      </AppTestProvider>
    )

  it('renders correctly with desktop projects', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders titles correctly', () => {
    renderComponent()

    desktopProjects.forEach(async () => {
      expect(await screen.findByText('Example mobile project')).toBeInTheDocument()
    })
  })

  it('renders correctly with mobile projects tools', async () => {
    renderComponent()

    desktopProjects.forEach((project) => {
      project.tools.forEach(async (tool) => {
        expect(await screen.getAllByLabelText(tool)).toBeTruthy()
      })
    })
  })

  it('displays the project image', () => {
    renderComponent()

    desktopProjects.forEach(async (project) => {
      expect(await screen.findByText(project.src)).toBeInTheDocument()
    })
  })
})
