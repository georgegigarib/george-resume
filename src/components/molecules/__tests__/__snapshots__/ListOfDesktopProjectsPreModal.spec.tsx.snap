// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListOfDesktopProjectsPreModal Component > renders correctly with desktop projects 1`] = `
<DocumentFragment>
  <div
    class="grid ml-8 grid-rows-1 grid-cols-4 
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
     border-l-0 rounded-l-none
    rounded-xl"
  >
    <div
      class="text-center mb-4 col-span-1 self-center p-2 mr-4 
        md:dark:bg-app-lightModeBoxes-box1 
    md:bg-white
    md:bg-opacity-20
    md:border-[2px]
    border-app-darkModeBoxes-box1
    dark:border-white
    dark:border-opacity-60
    md:shadow-md
    opacity-90 rounded-xl flex flex-col gap-4 mt-16"
    >
      <h3
        class="hidden md:flex text-lg font-bold dark:text-white text-black"
      >
        Example desktop project
      </h3>
      <div
        class="hidden md:flex flex-wrap justify-center gap-2"
      >
        <i
          aria-label="React"
          class="ci ci-React ci-lg"
          data-mui-internal-clone-element="true"
        />
        <i
          aria-label="TypeScript"
          class="ci ci-TypeScript ci-lg"
          data-mui-internal-clone-element="true"
        />
      </div>
    </div>
    <div
      class="col-span-3 self-center py-3 pr-3 ml-3 group"
    >
      <div
        class="swiper swiper-initialized swiper-vertical h-[120px] md:h-[250px] w-full"
      >
        <div
          class="swiper-wrapper"
          style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(0px, NaNpx, 0px);"
        >
          <div
            class="swiper-slide bg-transparent"
            data-swiper-slide-index="0"
          >
            <img
              alt="dev.projects.projects.desktop.exampleProject.title"
              class="rounded-2xl"
              src="desktop1.png"
            />
          </div>
          <div
            class="swiper-slide bg-transparent"
            data-swiper-slide-index="1"
          >
            <img
              alt="dev.projects.projects.desktop.exampleProject.title"
              class="rounded-2xl"
              src="desktop2.png"
            />
          </div>
          <div
            class="swiper-slide bg-transparent"
            data-swiper-slide-index="2"
          >
            <img
              alt="dev.projects.projects.desktop.exampleProject.title"
              class="rounded-2xl"
              src="desktop3.png"
            />
          </div>
        </div>
        <div
          class="swiper-pagination swiper-pagination-bullets swiper-pagination-vertical"
        >
          <span
            class="swiper-pagination-bullet"
          />
          <span
            class="swiper-pagination-bullet"
          />
          <span
            class="swiper-pagination-bullet swiper-pagination-bullet-active"
          />
        </div>
        <div
          class="hidden md:block"
        >
          <div
            class="autoplay-progress"
            slot="container-end"
          >
            <svg
              style="--progress: 0;"
              viewBox="0 0 48 48"
            >
              <circle
                cx="24"
                cy="24"
                r="20"
              />
            </svg>
            <span>
              3s
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
